### ISSUE TITLE
react-content-query-webpart Upgrading to latest drop v1.20

### ISSUE THREAD

**Issue by petecuttriss** [SharePoint/sp-dev-fx-webparts#308]
Thank you for reporting an issue or suggesting an enhancement. We appreciate your feedback - to help the team to understand your needs, please complete the below template to ensure we have the necessary details to assist you. 

#### Category
- [ ] Enhancement

Hi Simon-Pierre,

The [react-content-query-webpart sample](https://github.com/SharePoint/sp-dev-fx-webparts/tree/master/samples/react-content-query-webpart) that you've provided is a fantastic resource, showcasing a lot of really helpful custom property pane controls and techniques e.g. searching for sites/webs, selecting resulting lists, filtering on fields, ordering and selecting fields.

I've had a go at upgrading my local instance to the latest drop 1.2 which uses office-ui-fabric-react v4.32.0.   I found a couple things during this process so figured I'd share them, and what I did to get them working again.

1. The PropertyPaneAsyncDropdown wasn't reflecting the item that was selected from the dropdown list.  If you switched pages within the property pane, and then went back, it would show the correctly selected item.   To fix this I added a call to `this.context.propertyPane.refresh();` in the onCustomPropertyPaneChange of the ContentQueryWebPart.ts. 

2. The IPersonaProps interface no longer has a value property.  This causes an error for the CamlQueryHelper.ts and the ContentQueryService.ts.  I've changed this to instead use the primaryText property (not quite sure if this is correct or not, still testing).  So, in CamlQueryHelper.ts, line 163, I've changed it from x.value to x.primaryText.   In the ContentQueryService.ts file I've change line 580 to: `let isDuplicate = currentUsers.filter((u) => { return u.primaryText === user.primaryText; }).length > 0;`

3. Office Fabric v4.32.0 now renders the checkbox using a button, where previously it was using an input element.  The button doesn't support the 'data' attribute, so I've had to change the AsyncChecklist.tsx so that it returns the Checkbox using the 'value' inputProp rather than the 'data' inputProp.   I've also had to remove the 'data' property on it.

**office-ui-fabric-react Checkbox HTML previously**
![checkbox_before](https://user-images.githubusercontent.com/4825764/30191608-fa1320d2-9495-11e7-84c8-8c3287980e9c.JPG)

**office-ui-fabric-react Checkbox HTML v4.32.0**
![checkbox_after](https://user-images.githubusercontent.com/4825764/30191610-fd293126-9495-11e7-86b9-e79e475238f5.JPG)

Resulting checkbox code to return:
                `<Checkbox id={ item.id }
                          label={ item.label }
                          defaultChecked={ this.isCheckboxChecked(item.id) }
                          disabled={ this.props.disable }
                          onChange={ this.onCheckboxChange.bind(this) }
                          inputProps={ { value: item.id } }
                          className={ styles.checklistItem }
                          key={ index } />`

I've then also had to update line 43 in AsyncChecklist.tsx, so that it get's the 'value' rather than the 'data' attribute:
Previously:
`let checkboxKey = ev.currentTarget.attributes.getNamedItem('data').value;`

Now:
`let checkboxKey = ev.currentTarget.attributes.getNamedItem('value').value;`

 Thanks again for providing this sample, it's been a great help.  Hopefully the above helps.

Regards
Pete

**Comment by spplante** [comment]
Hi @petecuttriss, 

Thanks a lot for detailing the different steps you went through, I will surely use this in my next update. Just so you know, I recently fixed the **Web Url** by adding a **Site Url** parameter and changing the way it uses the search by using recursive calls in order to avoid being limited to 500 webs/sites. If you do not already have this version I would suggest you to take a look at it :)

My next update will be to pre-select the current site/web on which the user added the webpart, it will be a good opportunity to update to the latest drop, I will contact you here if ever I have questions :)

Thanks again!

**Comment by petecuttriss** [comment]
Hi @spplante,

My pleasure.  Glad I could help.

I did some further digging into the issue with the PropertyPaneAsyncDropdown not reflecting the selected item.   I ended up changing the code in the AsyncDropdown.tsx component.  The main change was altering the 'onChanged' binding.  I've bound it to a local method which in turn calls the prop.onChanged method.  It's now working happily.   I've attached the code.
[AsyncDropdown.tsx.txt](https://github.com/SharePoint/sp-dev-fx-webparts/files/1309549/AsyncDropdown.tsx.txt)

I'll grab your latest drop and take a look.

Thanks
Pete

**Comment by spplante** [comment]
Wonderful.

Based on when you took your version on here, you might not have the possibility to add external javascript files and custom handlebars block helpers, you can find how the feature works by looking at the new section all the way down in the documentation.

**Comment by spplante** [comment]
Hi @petecuttriss ,

I did upgrade the solution to 1.2.0 tonight, and I had the exact same issues as you. Here is what I figured:

1. I had the same problem with the PropertyPaneAsyncDropdown not showing the correct selected value. I did investiguate on this, and the issue [#2719](https://github.com/OfficeDev/office-ui-fabric-react/issues/2719) has already been raised on the office-ui-fabric repository. The issue was introduced in the 4.32.0 version, they did fix the issue already but the SharePoint packages are still pointing to the 4.32.0 version, hence why it is not fixed in SPFX yet. 

While your solution for this might be a good temporary fix, I feel like it's kind of an "hack", so I preferred raising the issue #311 to ask Microsoft to upgrade to a newer office-ui-fabric version. 

2. I also had the IPersona error with the .value being removed, however the solution is slightly different than replacing it with PrimaryText. The PrimaryText property is filled with the actual display name of the user and i used to display purpose only. On the other hand, the .value property was used to store the ID of the current user within the site collection. While using the display name for querying content might work, using the ID is critical in order to avoid conflicts and users with the same display name. So the line you were missing is line 375 in the ContentQueryService.ts, which I changed for this :

```javascript
                let userSuggestions:IPersonaProps[] = users.map((user) => { return { 
                    primaryText: user.DisplayText,
                    optionalText: user.EntityData.SPUserID || user.EntityData.SPGroupID
                }; });
```
So instead of storing the user ID in the value property, I stored it in the optionalText property for later use. I changed the other two occurences of .value as well.

3. I also had the problem and your solution was correct.

That's pretty much it, for now I will leave this "on hold" until the issue I raised is fixed, then as soon it's fixed I will publish this officially :)

Thanks again!

**Comment by petecuttriss** [comment]
Great work Simon-Pierre @spplante !  Thanks Pete.

**Comment by spplante** [comment]
Hi @petecuttriss,

I changed my mind and decided to go forward with a temporary solution that would fix the dropdown issue introduced in **office-ui-fabric-react@4.32.0**. I figured waiting on SPFX to update the package might take too long, so I went ahead and used most of your solution, except instead of storing the **selectedKey** in a private variable, I used the **state** in order be more faithfull to the react model and to fix some issues that I had when resetting the dropdown values later on...I also fixed some scss on the custom filter controls like the people picker and taxonomy picker.

I also took the occasion to add a small functionality that now automatically pre-select the current site url and web url on which the user adds the WebPart for the first time. This can be useful/quicker for 80% of the times where the user simply wants to query content on the current site.

I just submitted everything in the pull request #313, meanwhile if you want to start using it without waiting for the pull request to be merged, you can go get a version directly on the **dev** branch of my [fork](https://github.com/spplante/sp-dev-fx-webparts/tree/dev).

@VesaJuvonen, this enhancement can be closed 👍 

Thanks!

**Comment by spdevdocs** [comment]
We closed this issue as it had not activity within last 180 days. This is a generic process we have decided to perform for issues, which have not been explicitly marked still to be "work in progress" based on tags. We are performing this cleaning to make sure that old issues that have already been solved (but not closed) or are no longer relevant are cleaned out and make the issues more manageable. If this issue still valid, we would ask you to **open a new issue** and follow the guidance in the issue template related on the recommended location. We do apologize any inconveniences this might cause. Please do remember that issues in the issue lists are also messages for others in the community, so you can also check if you can assist on any of them. *“Sharing is caring!”*
